lib/internal/abort_controller.js

```diff
@@ -372,18 +374,51 @@ ObjectDefineProperty(AbortSignal.prototype, SymbolToStringTag, {
 
 defineEventHandler(AbortSignal.prototype, 'abort');
 
+// https://dom.spec.whatwg.org/#dom-abortsignal-abort
 function abortSignal(signal, reason) {
+  // 1. If signal is aborted, then return.
+  // if (signal[kReason]) signal[kAborted] == true;
   if (signal[kAborted]) return;
+
+  // 2. Set signal's abort reason to reason if it is given;
+  //    otherwise to a new "AbortError" DOMException.
   signal[kAborted] = true;
   signal[kReason] = reason;
+  // 3. Let dependentSignalsToAbort be a new list.
+  const dependentSignalsToAbort = [];
+  // 4. For each dependentSignal of signal's dependent signals:
+  if (signal[kDependantSignals]) {
+    SetPrototypeForEach(signal[kDependantSignals], (s) => {
+      const dependentSignal = s.deref();
+      // 1. If dependentSignal is not aborted, then:
+      if (dependentSignal && !dependentSignal[kAborted]) {
+        // 1. Set dependentSignal's abort reason to signal's abort reason.
+        dependentSignal[kReason] = reason;
+        dependentSignal[kAborted] = true;
+        // 2. Append dependentSignal to dependentSignalsToAbort.
+        ArrayPrototypePush(dependentSignalsToAbort, dependentSignal);
+      }
+    });
+  }
+  // 5. Run the abort steps for signal
+  runAbort(signal);
+  // 6. For each dependentSignal of dependentSignalsToAbort,
+  //    run the abort steps for dependentSignal.
+  for (let i = 0; i < dependentSignalsToAbort.length; i++) {
+    const dependentSignal = dependentSignalsToAbort[i];
+    runAbort(dependentSignal);
+  };
+}
+
+// To run the abort steps for an AbortSignal signal
+function runAbort(signal) {
+  // TODO: 1. For each algorithm of signal's abort algorithms: run algorithm.
+  // TODO: 2. Empty signal's abort algorithms.
```

Comment:
```suggestion
  // TODO(jazelly): 1. For each algorithm of signal's abort algorithms: run algorithm.
  // TODO(jazelly): 2. Empty signal's abort algorithms.
```

When adding TODO comments, putting the username of who added it can help with organization.